src/PROPOSAL/detail/PROPOSAL/scattering/stochastic_deflection/bremsstrahlung/TsaiParametrizationBremsstrahlungGinneken.cxx

```diff
@@ -0,0 +1,109 @@
+#include "PROPOSAL/scattering/stochastic_deflection/bremsstrahlung/TsaiParametrizationBremsstrahlungGinneken.h"
+#include "PROPOSAL/Constants.h"
+#include "PROPOSAL/math/MathMethods.h"
+#include <cmath> 
+#include <stdexcept>
+#include <algorithm>
+using namespace PROPOSAL;
+using namespace std;
+
+std::vector<double> logspace(double start, double end, int num)
+{
+
+  std::vector<double> linspaced;
+
+  if (num == 0) { return linspaced; }
+  if (num == 1) 
+    {
+      linspaced.push_back(start);
+      return linspaced;
+    }
+
+  double delta = (end - start) / (num - 1);
+
+  for(int i=0; i < num-1; ++i)
+    {
+      linspaced.push_back(pow(10, start + delta * i));
+    }
+  linspaced.push_back(pow(10, end)); // I want to ensure that start and end
+                            // are exactly the same as the input
+  return linspaced;
+}
+
+std::vector<double> f_nu_g(std::vector<double> nu, double n, double k_4) {
```

Comment:
I would suggest to implement the actual function standalone, not one that fills a vector.